classes/article/ArticleGalleyDAO.inc.php

```diff
@@ -286,17 +339,18 @@ function updateObject($galley) {
 	 * @param $galley ArticleGalley
 	 */
 	function deleteObject($galley) {
-		return $this->deleteById($galley->getId());
+		return $this->deleteById($galley->getId(), null, $galley->getCurrentVersionId());
```

Comment:
maybe to add articleId ($galley->getSubmissionId()) as well -- it is more safe...